From: Simon McVittie <smcv@debian.org>
Date: Sat, 21 May 2016 22:55:40 +0100
Subject: Try gtk-3.0 version of the icon cache utility first

GTK+ 3 in Debian used to install its icon cache updater as
gtk-update-icon-cache-3.0, to avoid colliding with gtk-update-icon-cache
from GTK+ 2. Support either way to facilitate backports.

Forwarded: no, Debian-specific
---
 triggers/gtk-icon-cache.trigger | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/triggers/gtk-icon-cache.trigger b/triggers/gtk-icon-cache.trigger
index 711cfab..fa780fe 100755
--- a/triggers/gtk-icon-cache.trigger
+++ b/triggers/gtk-icon-cache.trigger
@@ -1,11 +1,13 @@
 #!/bin/sh
 
-if test \( -x "$(which gtk-update-icon-cache 2>/dev/null)" \) -a \( -d $1/exports/share/icons/hicolor \); then
+update_icon_cache="$(command -v gtk-update-icon-cache-3.0 || command -v gtk-update-icon-cache || true)"
+
+if test \( -x "$update_icon_cache" \) -a \( -d $1/exports/share/icons/hicolor \); then
     cp /usr/share/icons/hicolor/index.theme $1/exports/share/icons/hicolor/
     for dir in $1/exports/share/icons/*; do
         if test -f $dir/index.theme; then
-            if ! gtk-update-icon-cache --quiet $dir; then
-                echo "Failed to run gtk-update-icon-cache for $dir"
+            if ! "$update_icon_cache" --quiet $dir; then
+                echo "Failed to run $update_icon_cache for $dir"
                 exit 1
             fi
         fi
